src/seedu/addressbook/data/person/Address.java

```diff
@@ -37,19 +85,20 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return String.format("%s, %s, %s, %s", 
+        		block.value, street.value, unit.value, postalCode.value);
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.state.equals(((Address) other).state)); // state check
```

Comment:
`state` is never assigned any value. It is not possible to compare as a result, and therefore is semantically incorrect.